### PR TITLE
Created BaseLogger

### LINKED ISSUE #8: Implement BaseLogger
Implement the abstract class, BaseLogger.

### PR DESCRIPTION
**Pull request by j3kstrum**
Created BaseLogger and implemented it in the EngineLogger.

Closes #8 

BaseLogger is a simple class. When instantiated, it has five methods - debug, info, warning, critical, and fatal. Each of which depicts a greater level of severity when logging. The simplest constructor takes in the name of the logger. More advanced constructors exist for further control.

### CODE REVIEW

**Review by seanbrais — CHANGES_REQUESTED** [review]
Lets put some comments in here.  @j3kstrum

**Comment by j3kstrum** [comment]
@seanbrais Comments included. Please review again.

File: src/common/BaseLogger.java [hunk 1]
```diff
@@ -3,53 +3,109 @@
...
 import java.time.LocalDateTime;
 
+/**
+ * A base class for logging. Can be instantiated to provide a quick and informative way of producing
+ * console or other stream output.
+ *
+ * @author jacob.ekstrum
+ */
 public class BaseLogger {
     private String _NAME;
```
Comment (seanbrais) [inline src/common/BaseLogger.java]: These should be camelcase

**Review by seanbrais — CHANGES_REQUESTED** [review]
Member variables should be lowercase

**Comment by j3kstrum** [comment]
@seanbrais fixed

**Review by seanbrais — APPROVED** [review]

**Pull request merged**